app/controllers/foreman_wreckingball/hosts_controller.rb

```diff
@@ -108,8 +108,27 @@ def triggering_params
       end
     end
 
-    def find_status
-      @status = HostStatus::Status.find_by!(:id => params[:status_id], :host_id => @host.id)
+    def statuses_params
+      @statuses_params ||= params.permit(:host_association, :owned_only, status_ids: [])
+    end
+
+    def find_statuses
+      @statuses = begin
+        host_association = statuses_params[:host_association]
+        status_class = HostStatus.find_wreckingball_status_by_host_association(host_association)
+        if status_class
+          Host.authorized(:remediate_vmware_status_hosts, Host)
+              .joins(status_class.host_association)
+              .includes(status_class.host_association)
+              .try { |query| statuses_params[:owned_only] ? query.owned_by_current_user_or_group_with_current_user : query }
+              .where.not('host_status.status': status_class.global_ok_list)
+              .map { |host| host.send(status_class.host_association) }
+        else
```

Comment:
Why do we need the else branch here?

Reply:
There are two ways to start remediate action: you can select hosts for remediation and then `status_ids` is passed in the params or you can remediate all hosts for status and then `host_association` and optionally `owner_only` are passed in the params.